cmd/krew/cmd/root.go

```diff
@@ -150,13 +150,8 @@ func preRun(cmd *cobra.Command, _ []string) error {
 	}
 
 	if _, ok := os.LookupEnv(constants.EnableMultiIndexSwitch); ok {
-		isMigrated, err = indexmigration.Done(paths)
-		if err != nil {
-			return errors.Wrap(err, "error getting file info")
-		}
-		if !isMigrated && cmd.Use != "index-upgrade" {
-			fmt.Fprintln(os.Stderr, "You need to perform a migration to continue using krew.\nPlease run `kubectl krew system index-upgrade`")
-			return errors.New("krew index outdated")
+		if err := indexmigration.Migrate(paths); err != nil {
+			return errors.Wrap(err, "failed to automatically migrate index")
```

Comment:
This error is no longer accurate. The index could already have been migrated, but the check failed. In this particular situation, I think it's best not to wrap at all, because the errors from `Migrate` are already descriptive. If you feel strongly about wrapping, I'd suggest something like
```suggestion
			return errors.Wrap(err, "index initialization")
```